lib/merge.js

```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+function merge (defaultObject, overrideObject) {
+  var newObject = JSON.parse(JSON.stringify(defaultObject))
+
+  for (var key in overrideObject) {
+    if (overrideObject.hasOwnProperty(key)) {
+      newObject[key] = overrideObject[key]
```

Comment:
This doesn't protect the attachments array in a payload from being manipulated after the fact, but since the scope of this module is pretty small, I don't think it's necessary to worry about that.